### PR TITLE
Add Prometheus and scrape metrics of Dashbase services. upgrade Grafana. Remove release name from service name

### PR DESCRIPTION
**Pull request by keitaf**
* Add Prometheus into the chart
* Add `dashbase.io/scrape` `dashbase.io/metric_path` and  `dashbase.io/scrape_port` annotations to other Pods so that Prometheus can scrape the metrics
* Upgrade Grafana to use Grafana 5 w/ Prometheus datasource
* Remove the release name from Service names so that services can be discovered just by their names. (Assumption: There is only one Dashbase deployment per K8s namespace)

### CODE REVIEW

File: chart/templates/kafka.yaml [hunk 1]
```diff
@@ -57,13 +57,17 @@ spec:
...
         component: kafka
+      annotations:
+        dashbase.io/scrape: "true"
+        dashbase.io/metric_path: /metrics
+        dashbase.io/scrape_port: "8080"
     spec:
       containers:
       - name: kafka
-        image: dashbase/kafka:{{ .Values.services.kafka.version | default "current" }}
+        image: "dashbase/kafka:{{ .Values.dashbase_version }}"
```
Comment (khou) [inline chart/templates/kafka.yaml]: we will unify version of kafka with dashbase? that may cause kafka to re-deploy when unnecessary though...

File: dashbase-k8s-solution/values.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-dashbase_version: 0.6.0
+dashbase_version: nightly
```
Comment (khou) [inline dashbase-k8s-solution/values.yaml]: not sure solution should use nightly version

File: chart/templates/kafka.yaml (hunk 1 above)
Comment (keitaf) [reply chart/templates/kafka.yaml]: I switched from `current` to `nightly` when migrating to Prometheus. i.e., `kafka:current` is using metricbeat to push metrics to our `_metric` table, whereas `kafka:nightly` exposes the metric endpoint for Prometheus to scrape.

we can (and probably should) set `imagePullPolicy: Always` to pull the latest image, and I'm hoping that k8s doesn't restart pod if the image SHA doesn't change (just like swarm), then it should be safe to use the versioned image I guess.

**Comment by keitaf** [comment]
* add alertmanager, cadvisor and influxdb (from add-prom branch)
* modify cadvisor to only deploy onto the nodes where other Dashbase service is running
* made necessary changes to make "Docker Container Monitoring" dashboard work on k8s (https://github.com/dashbase/dashbase-engine/pull/2654 https://github.com/dashbase/grafana-dashboards/pull/4)

File: chart/templates/kafka.yaml (hunk 1 above)
Comment (khou) [reply chart/templates/kafka.yaml]: gotcha

Comment (SinceJune) [reply chart/templates/kafka.yaml]: actually, this change may break the release deployment, cause there's no dashbase/kafka image tag 0.6.0

**Comment by SinceJune** [comment]
My updates:
- add pvc into grafana
- modify all services' ports to origin one ( API 80->9876), so that we won't mess them up.
- add service and some fix in certmanager

File: chart/templates/kafka.yaml (hunk 1 above)
Comment (keitaf) [reply chart/templates/kafka.yaml]: just pushed `dashbase/kafka:0.6.0`, which is same as `dashbase/kafka:current`

**Comment by keitaf** [comment]
> We are going to use the release name as namespace right?

Currently namespace is not specified in the chart. so you need to add `--namespace` when running `helm install`.

I guess we can put `{{.Release.Name}}` in the namespace of all resources. then `helm install --name XYZ` will instal the release `XYZ` into the namespace `XYZ`, which might be better. Let's do that in another PR.

> There's one thing I need to clarify, our filebeat.yml is installed to namespace kube-system only, if we have two release both have filebeat, it may bring some problems.

I'm deleting the namespace in filebeat.yaml so that the filebeat(s) will be deployed into the same namespace.

**Pull request merged**